modin/pandas/series_utils.py

```diff
@@ -296,6 +298,18 @@ def match(self, pat, case=True, flags=0, na=np.NaN):
         )
 
     def extract(self, pat, flags=0, expand=True):
+        regex = re.compile(pat, flags=flags)
+        # need an operator that can create more columns than before
+        if expand and regex.groups == 1:
```

Comment:
It's a pity that we couldn't use a low-level [`.explode()`](https://github.com/modin-project/modin/blob/b103735f05bc5cda2b19bf84f404e95b0bb894d5/modin/core/dataframe/pandas/dataframe/dataframe.py#L2049) here. Ideally, we'd like to do `._query_compiler._modin_frame.explode(axis=1, func=str_extract)`, however, the current behavior of [`.explode()`](https://github.com/modin-project/modin/blob/b103735f05bc5cda2b19bf84f404e95b0bb894d5/modin/core/dataframe/pandas/dataframe/dataframe.py#L2049) makes it impossible due to the strange logic of extending the meta-data across an opposite axis rather than an equal one.

https://github.com/modin-project/modin/blob/b103735f05bc5cda2b19bf84f404e95b0bb894d5/modin/core/dataframe/pandas/dataframe/dataframe.py#L2070-L2079

I believe we should question this behavior in a separate issue because it just doesn't look right to me. I'll create one shortly.